src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -122,11 +377,24 @@ where
 
     /// Cast a vote to change the set of validators.
     pub fn vote_for(&mut self, change: Change<N>) -> Result<Step<C, N>> {
+        info!(
+            "{:?}, {}being a validator, is voting for {:?}",
+            self.netinfo.our_id(),
+            if self.netinfo.is_validator() {
+                ""
+            } else {
+                "not "
+            },
+            change
+        );
```

Comment:
A non-validator should never vote, I think? In that case we should probably just return an error.